src/mlpack/tests/decision_tree_test.cpp

```diff
@@ -355,6 +592,26 @@ BOOST_AUTO_TEST_CASE(BasicConstructionTest)
 }
 
 /**
+ * Construct a tree with weighted labels.
+ */
+BOOST_AUTO_TEST_CASE(BasicConstructionTestWithWeight)
+{
+  arma::mat dataset(10, 1000, arma::fill::randu);
+  arma::Row<size_t> labels(1000);
+  arma::rowvec weights(labels.n_elem);
+  weights.ones();
+
+  for (size_t i = 0; i < 1000; ++i)
+    labels[i] = i % 3; // 3 classes.
+
+  // Use default parameters.
+  DecisionTree<> d(dataset, labels, 3, weights, 50);
+
+  // Now require that we have some children.
+  BOOST_REQUIRE_GT(d.NumChildren(), 0);
```

Comment:
Hi @rcurtin,sorry for the late reply. I haven't got the pointer here actually. Is that mean something like:
`BOOST_REQUIRE_EQUAL(d.NumChildren(), wd.NumChildren())`?